aggregation-worker/src/main/java/au/org/emii/aggregationworker/AggregationRunner.java

```diff
@@ -245,38 +300,97 @@ public void run(String... args) {
 
                 logger.info("Raw aggregated file size [" + outputFile.toFile().length() + " bytes]");
 
+                //  This map holds the WPS outputs to be returned in the ExecuteResponse
                 HashMap<String, String> outputMap = new HashMap<>();
 
                 //  Perform the conversion
                 //  Instantiate the correct converter for the requested mimeType & do the conversion
                 Converter converter = Converter.newInstance(resultMime);
+                convertedFile = jobDir.resolve("converted" + "." + converter.getExtension());
+
+                logger.info("Converting file to output Mime Type: " + resultMime);
 
-                convertedFile = jobDir.resolve("converted" + converter.getExtension());
                 converter.convert(outputFile, convertedFile);
 
+                //  Create a file manager for uploading files to S3
                 S3JobFileManager outputFileManager = new S3JobFileManager(outputBucketName, jobFileS3KeyPrefix, batchJobId);
-                String fullOutputFilename = outputFilename + "." + converter.getExtension();
-                outputFileManager.upload(convertedFile.toFile(), fullOutputFilename, resultMime);
 
+                //  Rename the converted file.
+                convertedFile = Files.move(convertedFile, jobDir.resolve(requestedOutputFilename + "." + converter.getExtension()));
 
-                String resultUrl = WpsConfig.getS3ExternalURL(outputBucketName,
-                        outputFileManager.getJobFileKey(fullOutputFilename));
+                //  Add the converted file to the zip file
+                zipContent.add(convertedFile.toFile());
 
                 //  URL for the status page for this job
                 String statusUrl = WpsConfig.getStatusServiceHtmlEndpoint(batchJobId);
 
+                //  Read the metadata record for the layer
+                String catalogueURL = WpsConfig.getProperty(GEONETWORK_CATALOGUE_URL_CONFIG_KEY);
+                String layerSearchField = WpsConfig.getProperty(GEONETWORK_CATALOGUE_LAYER_FIELD_CONFIG_KEY);
+                CatalogueReader catalogueReader = new CatalogueReader(catalogueURL, layerSearchField);
+                String metadataResponseXML = catalogueReader.getMetadataXML(layer);
```

Comment:
Longish method now - could the detail be moved to a helper method her?

Reply:
I'll have a look at what refactoring we can do